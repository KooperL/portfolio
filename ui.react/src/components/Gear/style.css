#Gear {
  display: flex;
  flex-shrink: 0;
}