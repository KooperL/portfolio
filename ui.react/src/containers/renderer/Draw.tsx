import { triangle } from "./utils/types";

export function DrawFigure(props: {name: string, h: number | string, w: number | string}) {
  // const displayGCD = gcd(+props.w, +props.h);
  return (
    <figure>
      {/* <svg viewBox={`${0-+props.w/2} ${0-+props.h/2} ${+props.w/2} ${+props.h/2}`} width={props.w} height={props.h}> */}
      <svg viewBox={`0 0 ${+props.w} ${+props.h}`} width={props.w} height={props.h}>
        <use xlinkHref={`#${props.name}`}/>
      </svg>
    </figure>
  );
}

export function drawBlueprint(props: {name: string, tris: triangle[]}) {
  return (
    <svg className="defs-only" xmlns="http://www.w3.org/2000/svg">
      <symbol id={props.name}>
        {props.tris.map((triangle, triangleIndex) => (
          <polygon key={triangleIndex}
            points={`${triangle.p[0].x},${triangle.p[0].y} ${triangle.p[1].x},${triangle.p[1].y} ${triangle.p[2].x},${triangle.p[2].y}`}
            style={{
              fill: triangle['c']['fill'] ?? 'red',
              stroke: triangle['c']['stroke'] ?? 'pink',
              strokeWidth: triangle['c']['strokeWidth'] || 5.0,
            }}
          />
          ))}
      </symbol>
    </svg>
  );
}