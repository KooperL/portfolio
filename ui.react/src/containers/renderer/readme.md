Pressing pause for now.
 TODO:
 - Vector transformation as a method of their class. This will let them keep their colour properties when undergoing a transformation
 - Edge plane clipping for simple raster
 - Try to do something to reduce duplicate code
 - Use ticker as switch to toggle different transformations for grid
 - Do the scale properly -> make it a method and take args
 - colour spectrum