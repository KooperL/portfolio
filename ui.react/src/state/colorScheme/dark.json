{
  "header": {
    "background": "#205375",
    "text": "#EFEFEF",
    "outline": "#EFEFEF"
  },
  "body": {
    "background": "#112B3C",
    "foreground": "#205375",
    "h1": "#F66B0E",
    "h2": "#EFEFEF",
    "text": "#F2F2F2"
  },
  "button": {
    "bgSolid": "#205375",
    "bgGrad1": "#455375",
    "bgGrad2": "#20535a",
    "text": "#F2F2F2"
  }
}
