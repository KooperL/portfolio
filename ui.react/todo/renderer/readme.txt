Pressing pause for now.
 TODO:
 - Vector transformation as a method of their class. This will let them keep their colour properties when undergoing a transformation
 - Edge plane clipping for simple raster
 - Try to do something to reduce duplicate code
 - Put engine components behind layers of abstraction to be changed without affecting core engine -> make it a method and take args
 - colour spectrum
 - switch between canvas, web-gl enabled canvas and svg rasterisation

Abandoning, performance is simply not going to be acceptable without spending more time learning optimising strategies. Time better spent implementing algo's with a public library. Shame.
KT to myself:
 - V1 had best performance, but function structure was a mess and wasn't designed to be a general engine
 - V2 Was more well made and class based, not optimal for react but function structure was very ideal. Probably where I should pick up
 - V3 Was V2 crammed into react functional components, eventually started impacting performance.