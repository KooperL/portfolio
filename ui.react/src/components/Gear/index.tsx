// @ts-ignore
import gear from "../../assets/gear.svg";
import './style.css'

export function Gear(props: {variant?: 'small' | 'large'}) {
  return (
    <div id="Gear" style={{width: props.variant==='small' ? 20 : 40, height: props.variant==='small' ? 20 : 40}}>
      <img src={gear} alt={gear} /> 
    </div>
  )
}