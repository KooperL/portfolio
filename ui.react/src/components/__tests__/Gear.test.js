import React from "react"
import renderer from "react-test-renderer"
import dark from "../../state/colorScheme/dark.json"
import { Gear } from "../Gear"

describe("Gear component", () => {
  test("it matches the snapshot", () => {
    const treeSmall = renderer.create(<Gear variant="small" />).toJSON()
    expect(treeSmall).toMatchSnapshot()
    // const treeLarge = renderer.create(<Gear variant='large' />).toJSON();
    // expect(treeLarge).toMatchSnapshot();
  })
})
