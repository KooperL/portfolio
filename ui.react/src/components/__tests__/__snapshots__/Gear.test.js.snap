// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gear component it matches the snapshot 1`] = `
<div
  id="Gear"
  style={
    Object {
      "height": 20,
      "width": 20,
    }
  }
>
  <img
    alt="gear.svg"
    src="gear.svg"
  />
</div>
`;
